### DISCUSSION TITLE
Use a reminder notification instead when not logged in upon client starting

### DISCUSSION THREAD

**Discussion by CloneWith** [ppy/osu#30376] [Ideas]
It's like bit of spam that the login panel always pops up when we launch osu!lazer without an account logged in, and can be quite distractive and annoying, since one may just want to have a try at lazer locally, or run local binaries for test purposes. Not to mention that online playing is actually an optional part of osu!lazer.

In these cases, I prefer using a notification instead of this popup, reading "You can access our online features once you are logged in. Click to open the account panel." or something like this.

**Comment by bdach** [comment]
How is a notification any less nagging than showing the login panel? I think the notification is as disruptive as the login panel showing up. If not more,.

**Reply by CloneWith** [reply]
I think the subtle difference is that when the login panel shows up, it would block user input to layers below it (which makes us click the blocked part or press the escape key to dismiss). On Android devices it also makes the soft keyboard appear and cover part of the screen (or push the screen upwards).

Compared to the popping panel, notifications are more friendly because it won't affect user's normal interaction with most components, and we can dismiss it simply with a swipe or click.

**Reply by bdach** [reply]
I dunno I prefer the subtle suggestion of not being logged in via the overlay showing up to being hammered with explicit "HEY LISTEN DID YOU LOG IN YET? REMEMBER THAT YOU CAN LOG IN" text every time I launch the game.

Maybe we can make the text boxes not automatically focus on mobile or something at most.

**Reply by peppy** [reply]
I think I'd prefer the notification. Can agree that the focus stealing is pretty annoying.

**Reply by Gabixel** [reply]
Both have their own drawbacks. Focus stealing is probably too much, and hard to quickly notice the focus change, but a subtle notification **at startup** is probably very easy to miss (and I'm personally not sure that notifications at the very beginning are an appreciable thing, with the exception for the update/changelog and background operations).

One idea might be something similar to what I saw in Muse Dash.
In that game, when you press "start" on the first map, a giant prompt steals your focus and suggests logging in to save scores and appear on the leaderboard, so in lazer it's probably better to just have a notification during the first beatmap load (similar to the "you hid the HUD" one), or maybe at the first carousel entrance? It might get more attention there.

I could be wrong, but I bring up this idea in case it might give some inspiration.

**Reply by CloneWith** [reply]
> One idea might be something similar to what I saw in Muse Dash.
In that game, when you press "start" on the first map, a giant prompt steals your focus and suggests logging in to save scores and appear on the leaderboard, so in lazer it's probably better to just have a notification during the first beatmap load (similar to the "you hid the HUD" one), or maybe at the first carousel entrance? It might get more attention there.

This seems to work probably better with the startup notification. For implementation, a Yes/No dialog can be used.

Are we considering it?
